src/linear_algebra/basis.lean

```diff
@@ -751,6 +751,63 @@ end
 
 end span
 
+lemma smul_group_span_eq_top
+  {G : Type*} [group G] [distrib_mul_action G R] [distrib_mul_action G M]
+  [is_scalar_tower G R M] {v : ι → M} (hv : submodule.span R (set.range v) = ⊤) {w : ι → G} :
+  submodule.span R (set.range (w • v)) = ⊤ :=
+begin
+  rw eq_top_iff,
+  intros j hj,
+  rw ← hv at hj,
+  rw submodule.mem_span at hj ⊢,
+  refine λ p hp, hj p (λ u hu, _),
+  obtain ⟨i, rfl⟩ := hu,
+  have : ((w i)⁻¹ • 1 : R) • w i • v i ∈ p := p.smul_mem ((w i)⁻¹ • 1 : R) (hp ⟨i, rfl⟩),
+  rwa [smul_one_smul, inv_smul_smul] at this,
+end
+
+/-- Given a basis `v` and a map `w` such that for all `i`, `w i` are elements of a group,
+`smul_group` provides the basis corresponding to `w • v`. -/
+def smul_group {G : Type*} [group G] [distrib_mul_action G R] [distrib_mul_action G M]
+  [is_scalar_tower G R M] [smul_comm_class G R M] (v : basis ι R M) (w : ι → G) :
+  basis ι R M :=
+@basis.mk ι R M (w • v) _ _ _
+  (linear_independent.group_smul v.linear_independent w) (smul_group_span_eq_top v.span_eq)
```

Comment:
Does dot notation work here?
```suggestion
  (v.linear_independent.group_smul w) (smul_group_span_eq_top v.span_eq)
```

Reply:
Doesn't work sadly

Reply:
Ok, on the second try it seems to work. Must of made a typo the first time around